File: GigRadio/Models/SongKick/SongKickDateTime.m
//
//  SongKickDateTime.m
//  GigRadio
//
//  Created by Michael Forrest on 16/08/2014.
//  Copyright (c) 2014 Good To Hear. All rights reserved.
//

#import "SongKickDateTime.h"
#import "DateFormats.h"
@implementation SongKickDateTime
+(NSDictionary *)defaultPropertyValues{
    return @{
             @"date": [NSDate dateWithTimeIntervalSince1970:0],
             @"datetime": [NSDate dateWithTimeIntervalSince1970:0],
             @"timeString": @""
             };
}
+(NSDictionary *)dictionaryConvertedFromJSON:(NSDictionary *)dict{
    NSMutableDictionary * result = dict.mutableCopy;
    if(dict[@"date"])
        result[@"date"] = [[DateFormats querystringDateFormatter] dateFromString:dict[@"date"]];
    
    if(dict[@"datetime"])
        result[@"datetime"] = [[DateFormats dateTimeFormat] dateFromString:dict[@"datetime"]];
//    else
//        result[@"datetime"] = [self createDateTimeFromDateAndTimeString];

    
    if(dict[@"time"])
        result[@"timeString"] = dict[@"time"];
    
    return result;
}
@end

File: GigRadio/Models/SongKick/SongKickDateTime.swift
//
//  SongKickDateTime.swift
//  GigRadio
//
//  Created by Michael Forrest on 17/05/2015.
//  Copyright (c) 2015 Good To Hear. All rights reserved.
//

import UIKit
import RealmSwift

class SongKickDateTime: Object {
    dynamic var datetime = ""
    dynamic var date = ""
    dynamic var time = "";
    class func dictionaryConvertedFromJSON(dict:NSDictionary)->NSDictionary{
        let result: NSMutableDictionary = dict.mutableCopy() as! NSMutableDictionary
        
        if let date = dict["date"] as? String{
           result["date"] = DateFormats.querystringDateFormatter().dateFromString(date)
        }
        if let datetime = dict["datetime"] as? String{
            result["datetime"] = DateFormats.dateTimeFormat().dateFromString(datetime)
        }
        if let time = dict["time"] as? String{
            result["timeString"] = dict["time"]
        }
        return result
    }
    
    func parsedDate()->NSDate?{
        if let result = DateFormats.querystringDateFormatter().dateFromString(date){
            return result
        }else if let result = DateFormats.dateTimeFormat().dateFromString(datetime){
            return result
        }else{
            return nil
        }
    }
}
